apigateway.go

```diff
@@ -17,40 +17,75 @@ import (
 	"strings"
 	"unicode"
 
+	"github.com/SpalkLtd/slogger"
+	"github.com/SpalkLtd/slogger/loggers/logentries"
+	"github.com/SpalkLtd/slogger/notifiers/airbrake"
 	apex "github.com/apex/go-apex"
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
 )
 
+const awsLambdaMaxBodySize = 6 * 1000 * 1000 // 6 MB
+
+var logger *slogger.SpalkLogger
+
+func init() {
+	env := os.Getenv("ENVIRONMENT")
+	if env == "" {
+		env = "dev"
+	}
+	logger = slogger.NewLogger()
+	logHost := os.Getenv("LS_LOG_HOST")
+	logToken := os.Getenv("LS_LOG_TOKEN")
+	if logHost != "" && logToken != "" {
+		// Note call depth offset here: should tell us which apig.RespondHTTP call caused any errors
+		leLogger, err := logentries.New(logHost, logToken, 0, nil, 1)
+		if err == nil {
+			logger.SetLogger(leLogger)
+		} else {
+			logger.SetDefaultLogger()
+			logger.WithError(err).Println()
+		}
+	} else {
+		logger.SetDefaultLogger()
+	}
+	logger.Logger.SetFlags(log.Ltime | log.Lmicroseconds | log.Lshortfile)
+
+	errbitHost := os.Getenv("ERRBIT_HOST")
+	errbitKey := os.Getenv("ERRBIT_API_KEY")
+	if errbitHost != "" && errbitKey != "" {
+		logger.Notifier = airbrake.New(errbitHost, errbitKey, env, 1234)
+	}
+	logger.SetTag("environment", env)
+}
+
 var ErrNoHandler = errors.New("No handler defined for event of that type")
 
 //Respond will produce a response that will get formatted such that apigateway will modify it's response to the browser
 func Respond(body interface{}, status int, req events.APIGatewayProxyRequest, err error) (events.APIGatewayProxyResponse, error) {
-	// log.Println("Entering Respond")
 	if body != nil && reflect.TypeOf(body).Kind() == reflect.Func {
-		log.Println("Unsuported return type")
+		logger.Println("Unsuported return type")
 	}
 	debug.PrintStack()
```

Comment:
This seems unnecessary and tres wasteful?

Reply:
The function check in general? Yup, it was already there when I got there, but I'm happy to remove it?